### DISCUSSION TITLE
Unable to connect to Home Assistant. Retrying in __ seconds -

### DISCUSSION THREAD

**Discussion by mikehanrahan** [homeassistant-apps/app-cloudflared#369] [Q&A]
### The problem

very similar to [Issue 263](https://github.com/brenner-tobias/addon-cloudflared/issues/263), I am unable to login to Home Assistant via external URL.  Internal URL remains functional.  No Reverse Proxy Manager involved.

My domain name was registered thru CloudFlare, and I use CloudFlare's DNS & Dynamic DNS.
I have uninstalled, re-installed and re-set this add-on many times, with the result always the same. I have been researching on HA community, reddit, and here on GitHub. I have pasted in the bits you have requested from others in this thread and the HA Community site.
I am happy to work with you in any way you wish to help resolve this.  Please let me know if you need any additional Info

### What version of Cloudflared has the issue?

4.0.10 (and earlier)

### What type of installation are you running?

Home Assistant OS

### Add-on YAML Configuration

```yaml
external_host_name: subdomain.domain.com     #(anonomized!)
additional_hosts: []
```

### Anything in the logs that might be useful for us?

```txt
HA Log:  
WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from *** (xxx.xxx.xxx.xxx). Requested URL: '/auth/token'.   (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36)

Cloudflared Startup Log: 
Add-on: Cloudflared
 Use a Cloudflare Tunnel to remotely connect to Home Assistant without opening any ports
 
 Add-on version: 4.0.10
 You are running the latest version of this add-on.
 System: Home Assistant OS 9.5  (aarch64 / raspberrypi4-64)
 Home Assistant Core: 2023.2.2
 Home Assistant Supervisor: 2023.01.1
 
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
 
[15:21:51] INFO: Checking add-on config...
[15:21:52] INFO: Checking for existing certificate...
[15:21:52] INFO: Existing certificate found
[15:21:52] INFO: Checking for existing tunnel...
[15:21:52] INFO: Existing tunnel with ID *** found
[15:21:52] INFO: Checking if existing tunnel matches name given in config
[15:21:53] INFO: Existing Cloudflare Tunnel name matches config, proceeding with existing tunnel file
[15:21:53] INFO: Creating config file...
[15:21:55] INFO: Validating config file...
Validating rules from /tmp/config.json
OK
[15:21:55] INFO: Creating DNS entry ***...
2023-02-07T20:21:56Z INF *** is already configured to route to your tunnel tunnelID=***
[15:21:56] INFO: Finished setting up the Cloudflare Tunnel
[15:21:57] INFO: Connecting Cloudflare Tunnel...
2023-02-07T20:21:57Z INF Starting tunnel tunnelID=***
2023-02-07T20:21:57Z INF Version 2023.2.1
2023-02-07T20:21:57Z INF GOOS: linux, GOVersion: go1.19.3, GoArch: arm64
2023-02-07T20:21:57Z INF Settings: map[config:/tmp/config.json cred-file:/data/tunnel.json credentials-file:/data/tunnel.json loglevel:info metrics:0.0.0.0:36500 no-autoupdate:true origincert:/data/cert.pem]
2023-02-07T20:21:57Z INF Generated Connector ID: 6157411c-9374-42d2-932e-6693ecf9a0fd
2023-02-07T20:21:57Z INF Initial protocol quic
2023-02-07T20:21:57Z INF ICMP proxy will use 172.30.33.3 as source for IPv4
2023-02-07T20:21:57Z INF ICMP proxy will use :: as source for IPv6
2023-02-07T20:21:57Z INF Starting metrics server on [::]:36500/metrics
2023-02-07T20:21:57Z INF Connection f6e0fd34-0c6c-49d1-939e-757759fd2121 registered with protocol: quic connIndex=0 ip=198.41.200.13 location=ORD
2023-02-07T20:21:58Z INF Connection 3d316fa8-cc94-444c-adf8-b330000e1955 registered with protocol: quic connIndex=1 ip=198.41.192.7 location=EWR
2023-02-07T20:21:59Z INF Connection f5f7a0cd-c35a-4b53-8093-d55f36c2b584 registered with protocol: quic connIndex=2 ip=198.41.192.67 location=EWR
2023-02-07T20:21:59Z INF Connection 99e7e1f2-abe9-43b1-8d3a-74285555027f registered with protocol: quic connIndex=3 ip=198.41.200.43 location=ORD
```

### Steps to reproduce the issue

1.  Access external url  via any device: android, desktop, etc, over  WiFi, LTE, etc
2.  See HA Login Page
3. Provide Username & password
4. Receive Error on screen and in log file
![HA-Login Error](https://user-images.githubusercontent.com/5025661/217380889-9769cfe1-44de-4eb0-9a95-b6a26fa1d31a.png)

**Comment by brenner-tobias** [comment]
Are you using any Cloudflare Access as additional authentification layer? Which browser are you using?

**Comment by mikehanrahan** [comment]
I use Chrome as a browser, and yes, have GitHub Authentication enabled on
Cloudflare. I have also tried with no Access control, but still getting the
same unsuccessful login result.

**Comment by maramazza** [comment]
I have the same issue with 2 different fresh HA installations

**Comment by mikehanrahan** [comment]
I just tried 2 more browsers (edge, duckduckgo), getting the same result

**Comment by jeefm** [comment]
Same here, HA unable to connect retrying xx seconds.  uninstalled and other recommandations done as well.

**Comment by Tekonsite** [comment]
I am having the same issue on my HomeAssistant.  Mine is a vm on Proxmox.  I had to do a restore back 2 days to the version before this recent update.  I now have no issues.  I turned off auto updates until this is resolved. 

Logger: homeassistant
Source: components/cloudflare/__init__.py:95
First occurred: 12:40:35 AM (9 occurrences)
Last logged: 8:40:35 AM

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/cloudflare/__init__.py", line 57, in update_records
    await _async_update_cloudflare(session, cfupdate, zone_id)
  File "/usr/src/homeassistant/homeassistant/components/cloudflare/__init__.py", line 95, in _async_update_cloudflare
    records = await cfupdate.get_record_info(zone_id)
  File "/usr/local/lib/python3.10/site-packages/pycfdns/__init__.py", line 105, in get_record_info
    record_information.append(CFRecord(data["result"][0]))
IndexError: list index out of range

**Comment by AlpineWhite** [comment]
> I am having the same issue on my HomeAssistant. Mine is a vm on Proxmox. I had to do a restore back 2 days to the version before this recent update. I now have no issues. I turned off auto updates until this is resolved.
> 
> Logger: homeassistant Source: components/cloudflare/**init**.py:95 First occurred: 12:40:35 AM (9 occurrences) Last logged: 8:40:35 AM
> 
> Error doing job: Task exception was never retrieved Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/components/cloudflare/**init**.py", line 57, in update_records await _async_update_cloudflare(session, cfupdate, zone_id) File "/usr/src/homeassistant/homeassistant/components/cloudflare/**init**.py", line 95, in _async_update_cloudflare records = await cfupdate.get_record_info(zone_id) File "/usr/local/lib/python3.10/site-packages/pycfdns/**init**.py", line 105, in get_record_info record_information.append(CFRecord(data["result"][0])) IndexError: list index out of range


I have the exact same symptoms and errors. 

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/cloudflare/__init__.py", line 57, in update_records
    await _async_update_cloudflare(session, cfupdate, zone_id)
  File "/usr/src/homeassistant/homeassistant/components/cloudflare/__init__.py", line 95, in _async_update_cloudflare
    records = await cfupdate.get_record_info(zone_id)
  File "/usr/local/lib/python3.10/site-packages/pycfdns/__init__.py", line 105, in get_record_info
    record_information.append(CFRecord(data["result"][0]))
IndexError: list index out of range

Was working well before.

**Comment by mikehanrahan** [comment]
Could a common thread be that we all have multiple domains under management in CloudFlare?

**Comment by AlpineWhite** [comment]
> Could a common thread be that we all have multiple domains under management in CloudFlare?

You are definitely onto something. I removed a site whose registration is expiring in 13 days and I was suddenly able to access my HA after going through the auth steps.

**Comment by mikehanrahan** [comment]
@maramazza @jeefm @Tekonsite - Do you all have multiple domains under management on CloudFlare?

**Comment by Tekonsite** [comment]
Yes I do but mine is fine now after I rolled back.




Sent from my Galaxy



-------- Original message --------
From: Mike H ***@***.***>
Date: 2023-02-13 10:50 a.m. (GMT-05:00)
To: brenner-tobias/addon-cloudflared ***@***.***>
Cc: Tekonsite ***@***.***>, Mention ***@***.***>
Subject: Re: [brenner-tobias/addon-cloudflared] Unable to connect to Home Assistant. Retrying in __ seconds - (Issue #318)


@maramazza<https://github.com/maramazza> @jeefm<https://github.com/jeefm> @Tekonsite<https://github.com/Tekonsite> - Do you all have multiple domains under management on CloudFlare?

**Comment by maramazza** [comment]
@mikehanrahan yes 4 domains

**Comment by jeefm** [comment]
no, I had 1 ***.tk
Jean-François Millaire

**Comment by nicopasla** [comment]
I have exactly the same problem, I have only one domain registered on Cloudflare,  My domain provider is Namecheap with a paid xyz domain

**Comment by matejlacko90** [comment]
I have also same problem, with one paid domain on the Cloudflare.
Cloudflare add-on logs are without errors. After I log in , the same screen appear (Unable to connect to Home Assistant).
Reason is that the login attempt came from my external IP address.

**Comment by masterfish1** [comment]
I started having this issue yesterday, no errors in the logs but if I go to domain.com, I get the "Unable to connect to Home Assistant" screen which, if left long enough, ends up on "400: Bad Request".

However, if I go to any of the subdomain.domain.com options I have configured, they all work fine.

**Comment by nicopasla** [comment]
It's fixed for me too, I just tried again this morning

**Comment by dragonflyuk** [comment]
I have found adding the trusted proxiess in the HTML section of HA solves the problem

`
http:
  server_port: 80
  use_x_forwarded_for: true
  trusted_proxies:
    - 173.245.48.0/20
    - 103.21.244.0/22
    - 103.22.200.0/22
    - 103.31.4.0/22
    - 141.101.64.0/18
    - 108.162.192.0/18
    - 190.93.240.0/20
    - 188.114.96.0/20
    - 197.234.240.0/22
    - 198.41.128.0/17
    - 162.158.0.0/15
    - 104.16.0.0/13
    - 104.24.0.0/14
    - 172.64.0.0/13
    - 131.0.72.0/22
    - 172.30.33.0/24
`

However I also discovered that the last line, was the IP address I was getting traffic from suggesting that cloud flares list is not uptodate. So if anyone has a more uptodate list, it is probably worth posting it, or a link to it.

**Comment by masterfish1** [comment]
I didn't have them in my configuration.yaml but did have 172.30.33.11. I changed that one to 172.30.33.0/24 and it worked.

I still don't understand why subdomain.domain.com would work but domain.com didn't beforehand but I care less now. Thanks @dragonflyuk

**Comment by dragonflyuk** [comment]
Further reading has suggested that the one I added, may infact be related to docker, rather than cloudflare.

**Comment by mikehanrahan** [comment]
I added all of the CloudFlare IPv4's to the trusted proxy list, restarted HA. No difference - still getting the very same error after providing HA credentials.  The "Login attempt Failed" notification on the HA dashboard says "Login attempt or request with invalid authentication from ..." sometimes shows an IPv4 address (not CloudFlare or Docker related), and sometimes (usually from a mobile device) shows a IPv6 address.

**Comment by johntdyer** [comment]
whats interesting is that if I refresh a few dozen times it starts working...

**Comment by Coedy** [comment]
I also have this issue purely from my work device (trying to tweak automations on my dinner break) It seems they use cloudflare too (198.41.200.XX appears in my addon's logs) I can access it via my phone (Sim data) and home computer (wifi locally but via my domain name). I just cant access it via my work devices presumably because they are using their own tunnelling? 
Being a non-network guy this is a bit of a nightmare for me haha.

**Comment by Liionboy** [comment]
same problem from today at me

**Comment by chrisverros** [comment]
Some problem for me right now. It was working perfect for the last week (from the initial setup until now)
Suddenly "Unable to connect to Home Assistant."
Resetting the service or even stop it and start it didn't solve the problem.
Restarting the whole HA vm did the trick and it is working.

**Comment by Liionboy** [comment]
At me i dont know how but now is work again

**Comment by dula34** [comment]
I have same problem. Connection problem resolved after restarting add-on. After 3 hours problem appeared again

**Comment by MartinSchillinger** [comment]
I get this error too. There must be some kind of authentication problem.

The Browser is doing a POST Request to /auth/token with this payload:
```
client_id: https://<tunnel domain>/
code: bb48c248239f4b0ba0e58eb1b0fca6b7
grant_type: authorization_code
```
which gets this response:
```
{"error":"invalid_request","error_description":"Invalid code"}
```

**Comment by Raftaman34** [comment]
I'm having this issue as well, appears to work fine on a network outside my house, such as mobile network. I am also unable to access any additional hosts internally as well.

**Comment by tamorgen** [comment]
I started having this problem just today.  No devices will connect externally and now I'm getting 400: bad request.  I've restarted HA, restarted Cloudflared, and cleared the cache on the CloudFlare management console (not HA).

**Comment by Coedy** [comment]
Interestingly. This doesn't seem to be an add-on issue, it seems to be an issue how home-assistant plays with cloudflare. 

I was fed up of this issue so tried disabling the add-on and installing the equivalent 'community add-on' on unraid (which is the same box that home assistant runs on in a VM) as I thought it would be easier for me to pass through other services as they'd be on the same 'level' as that add-on.

All other dockers get exposed perfectly (and still do). But home assistant shows the same issue after a week or two (sometimes only an hour or two) of using a new tunnel. I imagine it's the proxy system of home assistant that's fighting against cloudflare but I can't work out how to fix it. I've tried adding the cloudflare IPs to the 'http:' section of the home assistant config but still get the same.

**Comment by judahrand** [comment]
I suspect this has something to do with refresh tokens. Whenever I revoke them things resolve (for a while).

**Comment by dragonflyuk** [comment]
Can that be done remotely?

**Comment by elcajon** [comment]
I am moving this to the Discussions section as it is not a bug in the add-on itself.

**Comment by JBarzdo** [comment]
Any updates?

**Comment by charlesomer** [comment]
I am seeing a similar issue, macOS Safari shows the "Unable to connect" message, meanwhile Chrome works initially but after 30 seconds or so can no longer communicate with HA until the page is refreshed :(

**Comment by matteociaroni** [comment]
I had the same issue. I solved enabling websockets in the Cloudflare dashboard: https://developers.cloudflare.com/network/websockets/.

I noticed that the request to the websocket was failing with a 400, so I understand that something was going wrong with WSs.

**Comment by tarmacalastair** [comment]
Thanks for your answer about enabling websockets. I was using Cloudflare a while ago and couldn't get access to work so this week I tried using a [Tailscale tunnel from a VPS instead](https://medium.com/@mueez.khan/how-i-self-host-services-from-behind-cgnat-using-tailscale-and-a-vps-5ead3ccaa5c1). I then set up my proxies with [Nginx Proxy Manager (npm)](https://github.com/NginxProxyManager/nginx-proxy-manager) and got the same annoying message when I tried to log in. After reading your suggestion I enabled websockets support:

<img width="584" height="638" alt="ha-proxy" src="https://github.com/user-attachments/assets/a23505ad-22d3-4cc2-baaf-843469d491c5" />


And hey presto I was able to log in successfully!

Thanks!
